sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/aotcompat/Azure.Monitor.OpenTelemetry.Exporter.AotCompatibilityTestApp/Azure.Monitor.OpenTelemetry.Exporter.AotCompatibilityTestApp.csproj

```diff
@@ -0,0 +1,28 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFrameworks>net7.0;netstandard2.0</TargetFrameworks>
+    <PublishAot>true</PublishAot>
+    <TrimmerSingleWarn>false</TrimmerSingleWarn>
+    <IsTestSupportProject>true</IsTestSupportProject>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\..\..\core\Azure.Core\src\Azure.Core.csproj" />
```

Comment:
can consider using this syntax sugar: https://github.com/open-telemetry/opentelemetry-dotnet/blob/0b2d0367b29b397edaf2fcbddd9dd3c63611dfe1/test/OpenTelemetry.AotCompatibility.TestApp/OpenTelemetry.AotCompatibility.TestApp.csproj#L34C1-L35C69

Reply:
I think we can ignore this for now. I need the project reference here because Azure.Core has some unreleased annotations. I should be able to remove this line when Azure.Core releases their next monthly version.

Long Term, this repo will get an AOT CI job. https://github.com/Azure/azure-sdk-for-net/issues/38226

Reply:
Actually, we don't want the same trick as in OTel, at least not for all projects - Azure.Core is not trim compatible as a whole (and will take a while to get there - `DynamicData` for example is not workable as is). So we rely on the test app to only pick the pieces of Azure.Core actually needed by the OTel Exporter assembly.

OTel on the other hand is completely trim/AOT compatible as a whole - so it makes sense to "root" everything, to validate that fact.